Add support for downloading CSV file

bzfwunde opened the pull request:
Extended plugin to provide this, by adding the options:

import=<file to display>
export=:dokuwiki:media:path:where:to:write:mediafile.csv
linkname=Name of the download link to create

Thus, I changed the syntax to use '|' as delimiter between options
rather than using ' '.  I feel this is "more dokuwiki" :).  However,
this breaks existing dokuwiki pages that use the csv plugin to display
a CVS file, which now would have to replace the separating space (' ')
with '|import='.

The idea behind this extension is that I want to maintain a list (in
my case a member list) in dokuwiki but provide a download link to the
csv file (also for users with only read access to the page).  This is
now supported as shown in the following example:

<csv|export=:media:members.csv|linkname=Download member list>
Name,Email
,
Paul,paul@email.com
Andreads,andreas@dokuwiki.org
</csv>

It renders the CSV data as a DokuWiki table but adds a CSV download link
with the title "Download member list".

This is done in the following way:

When rendering the CSV data, this data is also copied to the export=
file.  Finally, a link to the just generated file is provided.

splitbrain commented:
In general I like the functionality but there's a couple of things I would change:

* keep spaces as separator. instead use optional double quotes around the parameter when you want to use spaces in the description 
* keep the file parameter to keep backwards compatibility (import could be an alias for it) 
* downloads should be supported for inline CSV data as well (see how downloadable code blocks do it)

bzfwunde commented:
On Tuesday 29 September 2015 08:52:23 Andreas Gohr wrote:
> In general I like the functionality but there's a couple of things I would
> change:
> 
> * keep spaces as separator. instead use optional double quotes around the
> parameter when you want to use spaces in the description

I understand the upside that this would not render it incompatible with
the previous version. That makes parsing of the parameters so much harder. I 
could no longer use explode and would actually have to start learning php :(. 
Or do you have a "-aware explode function awailable anywhere I could just use?

Further, isn't space separation really uncommon for dokuwiki plugins using the 
{{...}} syntax?

> * keep the file
> parameter to keep backwards compatibility (import could be an alias for it)

No problem with this one.

> * downloads should be supported for inline CSV data as well (see how
> downloadable code blocks do it)

I think it is working.  In fact, that's what I am using in my docuwiki 
installation, i.e. something like:

<csv|export=:media:members.csv|linkname=Download member list>
Name,Email
,
Paul,paul@email.com
Andreads,andreas@dokuwiki.org
</csv>

Or is that now what you mean?

Roland

bzfwunde commented:
OK, I changed to ' ' separators, wasn't so hard after all.
I also reverted back to file= instead of import=
I'm still not sure about your 3rd point.  IMO this just works.

splitbrain commented:
why did you close this?

bzfwunde commented:
Sorry - I only started today using github. This was my misuse of the tool.

bzfwunde commented:
Any news on pulling this?

bzfwunde commented:
Any reason why you don't want to pull this?  Are other changes needed?
